### PR TITLE
Admin page

### PR DESCRIPTION
**Pull request by julio1998**
Closes issues #9 #7 #6 #3

### CODE REVIEW

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/db/DbHelper.kt [hunk 1]
```diff
@@ -67,21 +77,31 @@ class DbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null
...
     /**
      * Removes the specified token from the DB
      */
-    fun removeToken(token: Token) {
+    fun removeToken(token: Token, admin: Boolean = false) {
+        var table = TABLE_TOKEN
+        if (admin) {
+            table = TABLE_ADMIN_TOKEN
+        }
+
```
Comment (Ne184) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/db/DbHelper.kt:98]: I think the admin wallet should not remove any tokens.
It should keep all previous versions of the tokens, right?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt [hunk 2]
```diff
@@ -4,40 +4,48 @@ import androidx.annotation.RequiresApi
...
+        tokens!!.add(token)
+        dbHelper?.addToken(token, admin = true)
     }
 
     fun removeToken(token: Token): Token{
-        tokens.remove(token)
-        dbHelper?.removeToken(token)
+        tokens!!.remove(token)
+        dbHelper?.removeToken(token, admin = true)
         return token
```
Comment (Ne184) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt]: Same comment, we should not remove token, correct me if I am wrong

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        tokens.remove(token)
+        dbHelper?.removeToken(token)
+        return token
+    }
+
+
+    // Return the number of tokens needed to pay the value
+    // For example: we need to pay 2 euros, but we have tokes of 0.50 cents, 1 euro token
+    // This means we need to return either two 1 euro tokens or 4x0.50 cents tokens
+    fun getPayment(value: Int): ArrayList<Token> {
```
Comment (Ne184) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt]: This function is not needed, I think we should remove it as well.

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt [hunk 4]
```diff
@@ -4,40 +4,48 @@ import androidx.annotation.RequiresApi
...
         private fun create(context: Context): AdminWallet {
 //            val generator = KeyPairGenerator.getInstance("RSA")
 //            generator.initialize(2048)
 //            val keyPair = generator.generateKeyPair()
 
-            val dbHelper = DbHelper(context)
-            val tokens = dbHelper.getAllTokens()
+            dbHelper = DbHelper(context)
+            tokens = dbHelper!!.getAllTokens(admin = true)
+            wallet = AdminWallet("s")
```
Comment (Ne184) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt:23]: Hehe, public key = 's'? Was there any reason to not put the same public key again?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt [hunk 5]
```diff
@@ -4,40 +4,48 @@ import androidx.annotation.RequiresApi
...
-        return tokens.size
+        return tokens!!.size
     }
 
     fun addToken(token : Token) {
-        tokens.add(token)
-        dbHelper?.addToken(token)
+        tokens!!.add(token)
+        dbHelper?.addToken(token, admin = true)
     }
```
Comment (Ne184) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt]: Do check whether the result was actually successfully written in the db with 
if(result != -1L) {
            tokens!!.add(token)
      } 
If it wasn't written successfully in the db, it should not be added to the tokens as well. You can look at the wallet class :)

**Review by Ne184 — CHANGES_REQUESTED** [review]
Really nice job overall, I have put a few comments. If you can take a look at them quickly, some of them are just remarks, for example the ones for the remove function. Feel free to merge in the main after you look at the comments.

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt (hunk 4 above)
Comment (julio1998) [reply detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt:23]: The admin doesn't have a key. I had trouble initiating the wallet without an input so I left this here. Not a really clean fix so might need some revising.

**Pull request merged**